### ISSUE TITLE
Empty value stops processing of layer data

### ISSUE THREAD

**Issue by tomass** [go-spatial/tegola#429]
It is useful to add some casing to polygon geometries (for example to water features). But as OpenStreetMap water polygons are made of parts (for various reasons), it could give unwanted artefacts as can be seen in this example where smaller river flows into the larger one:
![paveikslas](https://user-images.githubusercontent.com/969513/41195350-3444ca1e-6c34-11e8-870a-cdbdb1f75310.png)
The usual solution is to aggregate geometries with _st_union_ when creating vector tile (aggregating water geometries beforehand for whole region could create huge geometries so it is not a good solution). For example:
```
select
  gid,
  st_union(geom) AS geom
from
  planet_osm_polygon
where
  geom && !BBOX!
```
When a tile has no geometries in a BBOX, this produces a null value for geom (because _st_union_ is an aggregate function). Such null value stops Tegola processing of layer data with following output to console:

`2018/06/09 22:15:25 map.go:193: err fetching tile (z: 10, x: 584, y: 323) features: unable to decode layer (water10) geometry field (geom) into wkb where (gid = 1): EOF`

Now this would not be a problem (if we ignore error message), but if there is another part in a query:
```
select ... st_union(geom) AS geom ...
union all
select somethingelse AS geom from sometable
```
then geometries of the second part of such query are **not** placed into the vector tile basically leaving layer empty in particular tile.
I guess Tegola should handle such _null_ values as "no geometry" and continue with processing query results.

**Comment by ARolek** [comment]
@tomass thanks for the detailed report. I think your proposed solution should work fine. I will add it to our roadmap.

**Comment by ARolek** [comment]
This has been addressed in the v0.7.0 branch.

**Comment by tomass** [comment]
Can you reopen this one or should I create a new issue as originally I did not mention usage of _st_intersection_ which is causing "second part" of this problem?
(more info here: https://github.com/go-spatial/tegola/pull/430)

**Comment by ARolek** [comment]
@tomass re-opening.

**Comment by ARolek** [comment]
This fix has been merged into v0.9.x branch. I'm working on cutting a release soon.
